versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/DatabaseAdapterFactory.java

```diff
@@ -71,7 +71,8 @@ public Connector getConnector() {
       return connector;
     }
 
-    public abstract DatabaseAdapter build(ContentVariantSupplier contentVariantSupplier);
+    public abstract DatabaseAdapter build(
+        ContentVariantSupplier contentVariantSupplier, ContentTypeSupplier contentTypeSupplier);
```

Comment:
The number of params here is growing, at the same time we already have a pattern of "with" methods.... Would it make sense to move these suppliers to `withContentVariantSupplier` and `withContentTypeSupplier`?

Reply:
Yeah, this is a minor point. We can refactor later if it becomes a burden.